### ISSUE TITLE
Jetpack.com: testimonials section has accessibility issues

### ISSUE THREAD

**Issue by jkguidaven** [Automattic/jetpack#30055]
## What's the issue
In Jetpack.com, the testimonials section (present in the homepage) has some accessibility issues:

1. The section has no heading, which hampers its discoverability by screen reader users browsing the page through headings.
2. Navigating between testimonials with the arrow keys is not possible.
3. The link `Author / Investor / Podcaster` provides no context when seen in the list of all the page links by screen reader users.

## How to reproduce the issue
- Visit https://jetpack.com/
- Scroll to the testimonials section
- Notice the issues, using a screen reader of your choice

## Expected behavior

1. The document should contain an `h2` tag with content like `Testimonials`. If, for some reason, we don't want that heading to show on the page, we can [hide it while keeping it present for screen readers](https://css-tricks.com/inclusively-hidden/).
2. Navigating between testimonials should be possible using the arrow keys. The `Tab` key should move the focus from the testimonials slider to the next component, not to the next slide of the slider.
3. The link `Author / Investor / Podcaster` should have a more descriptive accessible name, such as `tim.blog` (using `aria-label`).

## Captures

_Testimonials section_
<img width="400" src="https://user-images.githubusercontent.com/1620183/227246135-148d3546-9f93-462c-8eba-32b6dcd7a7be.png" alt=""/>

**Comment by jkguidaven** [comment]
diff: D107916-code

**Comment by jkguidaven** [comment]
diff has been deployed

**Comment by andrii-lysenko** [comment]
I think this should be good to close.
